config/v1/techpreview.infrastructure.testsuite.yaml

```diff
@@ -12,3 +12,140 @@ tests:
       apiVersion: config.openshift.io/v1
       kind: Infrastructure
       spec: {}
+  onUpdate:
+  # platformStatus tests have to run in "onUpdate" because the Infrastructure has a
+  # status subresource, so we can't create an object with status.
+  - name: Should be able to find the default LoadBalancerType
+    initial: |
+      apiVersion: config.openshift.io/v1
+      kind: Infrastructure
+      spec:
+        platformSpec:
+          type: VSphere
+          vsphere: {}
+    updated: |
+      apiVersion: config.openshift.io/v1
+      kind: Infrastructure
+      spec:
+        platformSpec:
+          type: VSphere
+          vsphere: {}
+      status:
+        platform: VSphere
+        platformStatus:
+          vsphere: {}
+          type: VSphere
+    expected: |
+      apiVersion: config.openshift.io/v1
+      kind: Infrastructure
+      spec:
+        platformSpec:
+          type: VSphere
+          vsphere: {}
+      status:
+        controlPlaneTopology: HighlyAvailable
+        infrastructureTopology: HighlyAvailable
+        platform: VSphere
+        platformStatus:
+          vsphere:
+            loadBalancerType: OpenShiftManaged
+          type: VSphere
+  - name: Should be able to override the default LoadBalancerType with a valid value
+    initial: |
+      apiVersion: config.openshift.io/v1
+      kind: Infrastructure
+      spec:
+        platformSpec:
+          type: VSphere
+          vsphere: {}
+    updated: |
+      apiVersion: config.openshift.io/v1
+      kind: Infrastructure
+      spec:
+        platformSpec:
+          type: VSphere
+          vsphere: {}
+      status:
+        platform: VSphere
+        platformStatus:
+          vsphere:
+            loadBalancerType: UserManaged
+          type: VSphere
+    expected: |
+      apiVersion: config.openshift.io/v1
+      kind: Infrastructure
+      spec:
+        platformSpec:
+          type: VSphere
+          vsphere: {}
+      status:
+        controlPlaneTopology: HighlyAvailable
+        infrastructureTopology: HighlyAvailable
+        platform: VSphere
+        platformStatus:
+          vsphere:
+            loadBalancerType: UserManaged
+          type: VSphere
+  - name: Should not allow changing the immutable LoadBalancerType field
+    initial: |
+      apiVersion: config.openshift.io/v1
+      kind: Infrastructure
+      spec:
+        platformSpec:
+          type: VSphere
+          vsphere: {}
+      status:
+        controlPlaneTopology: HighlyAvailable
+        infrastructureTopology: HighlyAvailable
+        platform: VSphere
+        platformStatus:
+          vsphere:
+            loadBalancerType: OpenShiftManaged
+          type: VSphere
```

Comment:
The comment at the start of this section says you can't set this here and in the tests above they are set in 2 operations. Yet this apparently works. Why the inconsistency?

Reply:
The `onUpdate` tests handle this specially (it's very much a semantic to make sure people realise it's an update happening not a create operation), so they will run a status update after create to make sure the initial object status is populated (yes, there are some behavioural differences between a create and update especially around CEL validation), so it's ok to have the status in the initial, only on `onUpdate` tests though

This is useful if you want to check a validation for a field transition, otherwise you'd need multiple update runs to check ok, I can set it from empty to something, then from something to something else